python/cudf/cudf/tests/utils.py

```diff
@@ -113,6 +115,73 @@ def assert_neq(left, right, **kwargs):
         raise AssertionError
 
 
+def assert_exceptions_equal(
+    reference_func,
+    actual_func,
+    reference_func_arg_tuple=None,
+    actual_func_arg_tuple=None,
```

Comment:
Do these names sound better? It's more useful to describe what the argument needs to be rather than its type (`tuple`). The type is already available in the `Parameters` section of the docstring.

```suggestion
lfunc,
rfunc,
lfunc_args_and_kwargs,
rfunc_args_and_kwargs,
compare_error_message
```

Reply:
Probably best not to accept this suggestion since you also need to change the function body if you accept it :)

Reply:
Renamed the variables here, but I think keeping the default parameters as `None` for `args_and_kwargs` would be useful as that can allow us to pass in just any two callable functions and compare their assertions.